### DISCUSSION TITLE
How to create HA environment with cluster-api

### DISCUSSION THREAD

**Discussion by satishdotpatel** [kubernetes-sigs/cluster-api#9931] [Q&A] [answered]
I have deployed kubespray 3 multi-node cluster for HA environment and I have deployed CAPI on that cluster using `clusterctl init` command. by default it used replica of 1/1 for all deployments. My question is how do I create replica of 3 for high availability and how does data replication work? I am new to k8s so asking very basic question so help me out to create fully HA environment where any node can go down without impacting CAPI functionality. 

Does CAPI k8s deployment store any data for workload cluster, Do I need to keep taking backup regularly or its just control plane and nothing to take backup? 

Also how does backup/restore work with CAPI?

**Comment by satishdotpatel** [comment]
Can someone put some light on this topic?

**Comment by fabriziopandini** [comment, marked as answer]
This is a very broad question, and I fear it will be unlikely that someone can invest time in such a broad knowledge bump.
I would suggest to should look for ways to eat the elefant in smaller bits...

Some pointers:
- it is correct, CAPI usually doesn't ship HA ready manifest, you have to build your own; see e.g https://cluster-api.sigs.k8s.io/clusterctl/configuration.
- CAPI store data in etcd, and etcd takes care of replication; it also defines how much control plane nodes can go down before you have a disaster (this is actually Kubernetes HA, not Cluster API specific)
- You have to use backups or rely on practices like gitops to be able to recover in case of disasters
